scripts/irods/test/resource_suite.py

```diff
@@ -513,68 +513,80 @@ def test_local_iput_onto_specific_resource(self):
         if os.path.exists(datafilename):
             os.unlink(datafilename)
 
-    @unittest.skipIf(True, 'Enable once #2634 is resolved')
     def test_local_iput_interrupt_directory(self):
-        # local setup
-        datadir = "newdatadir"
-        os.makedirs(datadir)
+        local_dir_path = os.path.join(self.admin.local_session_dir, "newdatadir")
+        collection_path = os.path.join(self.admin.session_collection, 'newdatadir')
         datafiles = ["file1", "file2", "file3", "file4", "file5", "file6", "file7"]
+        restartfile = os.path.join(self.admin.local_session_dir, 'collectionrestartfile')
+        iputcmd = f'iput -X {restartfile} -r {local_dir_path} {collection_path}'
+
+        os.makedirs(local_dir_path)
         for datafilename in datafiles:
             print("-------------------")
             print("creating " + datafilename + "...")
-            localpath = datadir + "/" + datafilename
+            localpath = local_dir_path + "/" + datafilename
             lib.make_file(localpath, 2**20)
-        restartfile = "collectionrestartfile"
-        # assertions
-        iputcmd = "iput -X " + restartfile + " -r " + datadir
-        if os.path.exists(restartfile):
-            os.unlink(restartfile)
-        self.admin.interrupt_icommand(iputcmd, restartfile, 10)  # once restartfile reaches 10 bytes
-        assert os.path.exists(restartfile), restartfile + " should now exist, but did not"
+
+        # once restartfile reaches 10 bytes
+        self.admin.interrupt_icommand(iputcmd, restartfile, 10)
+        # Wait for all interrupted threads to exit - the files are large enough to trigger parallel transfer and some
+        # suites that run this test may be running some replication, sync-to-archive, etc.
+        # TODO: #6879 - This sleep will not be necessary if iput disconnects on SIGTERM.
+        time.sleep(10)
+
+        self.assertTrue(os.path.exists(restartfile), msg=f'{restartfile} should now exist, but did not')
+
         print("  restartfile [" + restartfile + "] contents --> [")
         with open(restartfile, 'r') as f:
             for line in f:
                 print(line)
         print("]")
-        self.admin.assert_icommand("ils -L " + datadir, 'STDOUT_SINGLELINE', datadir)  # just to show contents
-        self.admin.assert_icommand(iputcmd, 'STDOUT_SINGLELINE', "File last completed")  # confirm the restart
-        for datafilename in datafiles:
-            self.admin.assert_icommand("ils -L " + datadir, 'STDOUT_SINGLELINE', datafilename)  # should be listed
-        # local cleanup
-        lib.execute_command(['rm', '-rf', datadir])
-        if os.path.exists(restartfile):
-            os.unlink(restartfile)
 
-    @unittest.skipIf(True, 'Enable once race conditions fixed, see #2634')
+        _,out,_ = self.admin.assert_icommand(['ils', '-L', collection_path], 'STDOUT', collection_path)
```

Comment:
If you want, for `assert_icommand(...)` you can throw in a `desired_rc=<number>` for a bonus check.

Reply:
Done